quickwit/quickwit-config/src/index_config/serialize.rs

```diff
@@ -58,63 +59,51 @@ pub fn load_index_config_from_user_config(
 ) -> anyhow::Result<IndexConfig> {
     let versioned_index_config: VersionedIndexConfig = config_format.parse(config_content)?;
     let index_config_for_serialization: IndexConfigForSerialization = versioned_index_config.into();
-    index_config_for_serialization.validate_and_build(Some(default_index_root_uri))
+    index_config_for_serialization.build_and_validate(Some(default_index_root_uri))
 }
 
 impl IndexConfigForSerialization {
     fn index_uri_or_fallback_to_default(
         &self,
         default_index_root_uri_opt: Option<&Uri>,
     ) -> anyhow::Result<Uri> {
-        if let Some(index_uri) = self.index_uri.as_ref() {
+        if let Some(index_uri) = &self.index_uri {
             return Ok(index_uri.clone());
         }
         let default_index_root_uri = default_index_root_uri_opt.context("missing `index_uri`")?;
         let index_uri: Uri = default_index_root_uri.join(&self.index_id)
             .context("failed to create default index URI. this should never happen! please, report on https://github.com/quickwit-oss/quickwit/issues")?;
         info!(
-            index_id = %self.index_id,
-            index_uri = %index_uri,
-            "Index config does not specify `index_uri`, falling back to default value.",
+            index_id=%self.index_id,
+            index_uri=%index_uri,
+            "index config does not specify `index_uri`, falling back to default value",
         );
         Ok(index_uri)
     }
 
-    pub fn validate_and_build(
+    pub fn build_and_validate(
         self,
         default_index_root_uri: Option<&Uri>,
     ) -> anyhow::Result<IndexConfig> {
-        validate_identifier("Index ID", &self.index_id)?;
+        validate_identifier("index", &self.index_id)?;
```

Comment:
question: do we want to validate the identifier in `validate_index_config` (I don't have a strong opinion on that)

Reply:
(I just saw the usage in the template validation function, so yes, we want to keep the identifier validation outside this function)